### PR TITLE
doc: add @pkg-install alias documentation

### PR DESCRIPTION
**Pull request by maiste**
This PR adds the documentation for #11046.

### CODE REVIEW

**Review by Leonidas-from-XIV — COMMENTED** [review]
Nice. I think it would be also worth going through the explanation or the tutorial document for package management and add a link to this in relevant places, so people are aware of it (e.g. for regular users to know it and not only those that look at the documentation of every alias) and the documentation is better interlinked.

File: doc/reference/aliases/pkg-install.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@pkg-install`` will fetch the dependencies described in the ``depends`` field
+of your ``dune-project`` (see :doc:`/reference/dune-project/package`) and build them.
+
+.. note::
+    Note it will not build your project, only its dependencies. It is useful if
+    you are building projects using per-layer caching systems, e.g. Docker
+    images.
+
+If you are building the ``@pkg-install`` alias in a repository where package
+management is not activated, the command has no effect.
```
[the remark this answers is not in the document]
Comment (maiste) [reply doc/reference/aliases/pkg-install.rst:15]: Oh, good catch!

[the remark this answers is not in the document]
Comment (Leonidas-from-XIV) [reply doc/reference/aliases/pkg-install.rst:21]: I'm ok with either.

**Comment by maiste** [comment]
I emphasize in the explanation that you can use `dune build` if you want to build the project **and** download the deps, and add a note in the _tutorial_ about package management.

**Review by christinerose — APPROVED** [review]

File: doc/tutorials/dune-package-management/setup.md [hunk 2]
```diff
@@ -100,6 +100,13 @@ $ dune exec ./test.exe
 Hello, OCaml, Rust!
 ```
 
+:::{note}
+If you want to only build and fetch the project dependencies, you can use
+`@pkg-install` build alias.
```
Comment (Leonidas-from-XIV) [inline doc/tutorials/dune-package-management/setup.md:105]: Nice and good to link up the documentation of the alias, though I would make this into an example, that way people don't have to think what a build alias is:

```suggestion
the `@pkg-install` alias like so

$ dune build @pkg-install
```

(with the right markup)

**Review by Leonidas-from-XIV — COMMENTED** [review]
Looks good, just one more detail and then it's ready to merge.

**Review by Leonidas-from-XIV — COMMENTED** [review]
I hoped rebasing would fix the macOS issue, but no, it happens now on `main` as well.

The text looks good to me, can be merged.

**Review by Leonidas-from-XIV — APPROVED** [review]

**Pull request merged**